### PR TITLE
[CDAP-12823] Ensure DynamicPartitionerWriterWrapper creates new RecordWriter

### PR DESCRIPTION
**Pull request by ecfm**
https://issues.cask.co/browse/CDAP-12823
https://builds.cask.co/browse/CDAP-DUT6183-3
Ensure DynamicPartitionerWriterWrapper uses a new FileOutputFormat to create a new RecordWriter

### CODE REVIEW

**Comment by anwar6953** [comment]
Can we then remove the workaround in the plugin?

**Comment by ecfm** [comment]
@anwar6953 yes, we can remove it in a new version of the plugin. We still need to keep it in an older version of the plugin to be compatible with existing cdap.

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/dataset/partitioned/DynamicPartitionerWriterWrapper.java [hunk 1]
```diff
@@ -102,7 +104,7 @@ TaskAttemptContext getKeySpecificContext(PartitionKey partitionKey) throws IOExc
    * @throws IOException
    */
   RecordWriter<K, V> getBaseRecordWriter(TaskAttemptContext job) throws IOException, InterruptedException {
-    return fileOutputFormat.getRecordWriter(job);
+    return createFileOutputFormat().getRecordWriter(job);
```
Comment (anwar6953) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/dataset/partitioned/DynamicPartitionerWriterWrapper.java]: maybe put a comment, referencing the JIRA, explaining why we can't cache/reuse the FileOutputFormat?

**Review by anwar6953 — APPROVED** [review]

**Comment by anwar6953** [comment]
LGTM. Just squash commits

**Pull request merged**